apps/meteor/ee/server/patches/verifyContactChannel.ts

```diff
@@ -23,31 +24,51 @@ export const runVerifyContactChannel = async (
 		throw new Error('error-invalid-room');
 	}
 
-	await LivechatContacts.updateContactChannel(
-		{
-			visitorId,
-			source: room.source,
-		},
-		{
-			verified: true,
-			verifiedAt: new Date(),
-			field,
-			value: value.toLowerCase(),
-		},
-	);
+	const session = client.startSession();
+	let result = null;
+	try {
+		session.startTransaction();
+		await LivechatContacts.updateContactChannel(
+			{
+				visitorId,
+				source: room.source,
+			},
+			{
+				verified: true,
+				verifiedAt: new Date(),
+				field,
+				value: value.toLowerCase(),
+			},
+			{},
+			{ session },
+		);
 
-	await LivechatRooms.update({ _id: roomId }, { $set: { verified: true } });
+		await LivechatRooms.update({ _id: roomId }, { $set: { verified: true } }, { session });
 
-	const mergeContactsResult = await mergeContacts(contactId, { visitorId, source: room.source });
+		const mergeContactsResult = await mergeContacts(contactId, { visitorId, source: room.source }, session);
 
+		await session.commitTransaction();
+
+		result = mergeContactsResult;
+	} catch (error) {
+		console.log(error);
```

Comment:
Mongo transactions are weird so you should retry them at least once when having a transient error